contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -3112,6 +3119,309 @@ exec_stmt_insert_bulk(PLtsql_execstate *estate, PLtsql_stmt_insert_bulk *stmt)
 	return PLTSQL_RC_OK;
 }
 
+
+int exec_stmt_dbcc(PLtsql_execstate *estate, PLtsql_stmt_dbcc *stmt)
+{
+	switch (stmt->dbcc_stmt_type)
+	{
+		case PLTSQL_DBCC_CHECKIDENT:
+			exec_stmt_dbcc_checkident(stmt);
+			break;
+		default:
+			Assert(0);
+	}
+	return PLTSQL_RC_OK;
+}
+
+
+void exec_stmt_dbcc_checkident(PLtsql_stmt_dbcc *stmt)
+{
+	struct dbcc_checkident dbcc_stmt = stmt->dbcc_stmt_data.dbcc_checkident;
+	Relation	rel;
+	TupleDesc	tupdesc;
+	char	*db_name = NULL;
+	char	*schema_name = NULL;
+	char	*nsp_name = NULL;
+	char	*table_name;
+	char	*query;
+	char	*attname;
+	const char	*user;
+	const char	*guest_role_name;
+	int64	max_identity_value = 0;
+	int64	cur_identity_value = 0;
+	char	*max_identity_value_str = NULL;
+    char	*login = GetUserNameFromId(GetSessionUserId(), false);
+	char	*token;
+	int		attnum;
+	int		rc;
+	int64	reseed_value = 0;
+	Oid		nsp_oid;
+	Oid 	table_oid;
+	Oid		seqid = InvalidOid;
+	Oid		current_user_id = GetUserId();
+	volatile bool cur_value_is_null = true;
+	bool	login_is_db_owner;
+	char		message[200];
+
+
+    bool    is_cross_db = false;
+	if(dbcc_stmt.new_reseed_value)
+	{
+		/* If float value is passed as reseed_value, only decimal part is considered.*/
+		token = strtok(dbcc_stmt.new_reseed_value, ".");
+		reseed_value = pg_strtoint64(token);
+		pfree(token);
+	}
+
+    db_name = get_cur_db_name();
+    if (dbcc_stmt.db_name)
+    {
+        if (pg_strncasecmp(db_name, dbcc_stmt.db_name, NAMEDATALEN) != 0)
+            is_cross_db = true;
+        db_name = pstrdup(dbcc_stmt.db_name);
+    }
+
+	if (!DbidIsValid(get_db_id(db_name)))
+	{
+		ereport(ERROR,
+		(errcode(ERRCODE_UNDEFINED_DATABASE),
+			errmsg("database \"%s\" does not exist", db_name)));
+		pfree(db_name);
+	}
+
+	login_is_db_owner = 0 == strncmp(GetUserNameFromId(GetSessionUserId(), false),
+										get_owner_of_db(db_name), NAMEDATALEN);
+	user = get_user_for_database(db_name);
+	guest_role_name = get_guest_role_name(db_name);
+
+    if(is_cross_db)
+    {
+        if (user)
+            SetCurrentRoleId(GetSessionUserId(), false);
+        else
+            ereport(ERROR,
+                    (errcode(ERRCODE_UNDEFINED_DATABASE),
+                     errmsg("The server principal \"%s\" is not able to access "
+                            "the database \"%s\" under the current security context",
+                            login, dbcc_stmt.db_name)));
+    }
+    pfree(login);
+
+	if (dbcc_stmt.schema_name)
+	{
+		schema_name = pstrdup(dbcc_stmt.schema_name);
+		nsp_name = get_physical_schema_name(db_name, schema_name);
+	}
+	else
+	{
+		/* 
+		 * If schema_name is not provided, find default schema for current user
+		 * and get physical schema name
+		 */
+		if (!user)
+		{
+			ereport(ERROR,
```

Comment:
I think adding test-case for this error is not possible. `get_user_for_database(db_name)` will check if the login is the db owner, or login is member of sysadmin, or guest is enabled in login. Otherwise it will return NULL. This case we have covered under corss_db calls. And in cross-db calls when user is NULL we will throw error `The server principal \"%s\" is not able to access the database \"%s\" under the current security context`

Reply:
No, user can  not be null without cross-db calls.  
For example we can see that api  `get_user_for_database(db_name)` will return NULL when we are trying to do a cross db call and there is no user present for that db for the current login. Adding an assert for debugging.